tests/components/ecovacs/test_sensor.py

```diff
@@ -152,3 +152,34 @@ async def test_disabled_by_default_sensors(
         ), f"Entity registry entry for {entity_id} is missing"
         assert entry.disabled
         assert entry.disabled_by is er.RegistryEntryDisabler.INTEGRATION
+
+
+@pytest.mark.usefixtures(
+    "entity_registry_enabled_by_default", "mock_vacbot", "init_integration"
+)
+@pytest.mark.parametrize(("device_fixture"), ["123"])
+async def test_legacy_sensors(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+    mock_vacbot,
+) -> None:
+    """Test that sensor entity snapshots match."""
+    mock_vacbot.components = {"main_brush": 0.8, "side_brush": 0.6, "filter": 0.4}
+    mock_vacbot.lifespanEvents.notify("dummy_data")
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    states = hass.states.async_entity_ids()
+    assert snapshot(name="states") == states
+
+    for entity_id in hass.states.async_entity_ids():
+        assert (state := hass.states.get(entity_id)), f"State of {entity_id} is missing"
+        assert snapshot(name=f"{entity_id}:state") == state
```

Comment:
Considered using `snapshot_platform` helper?

Reply:
these tests are adopted from the tests above, therefor i didn't want to change their structure to be aligned to the other tests in this test module